httpcore/_sync/http2.py

```diff
@@ -314,17 +314,25 @@ def send_headers(
     ) -> None:
         scheme, hostname, port, path = url
         default_port = {b"http": 80, b"https": 443}.get(scheme)
-        if port is None or port == default_port:
+
+        authority = None
```

Comment:
Let's consider adding a code comment to explain what's going on here — I think it can be helpful to remind the reader that HTTP/2 doesn't use `Host` but has `:authority`.

```suggestion
        # In HTTP/2 the ':authority' pseudo-header is used instead of 'Host'.
        # For convenience we automatically treat any 'Host' header value
        # as a value for ':authority', allowing users to pass 'Host' whenever they
        # want to customize the host header regardless of the HTTP version in use.
        authority = None
```